rcl_lifecycle/src/default_state_machine.c

```diff
@@ -697,6 +697,7 @@ rcl_lifecycle_init_default_state_machine(
   // Concatenate the error strings if that happens
   if (rcl_error_is_set()) {
     fail_error_message = rcl_get_error_string().str;
+    rcl_reset_error();
```

Comment:
fail_error_message is only pointer.  
rcl_get_error_string().str points to global variable `gtls_rcutils_error_string`.  
After rcl_reset_error(), str of `gtls_rcutils_error_string` is set to “\0".  
So fail_error_message will point to empty.  
If want to fix  message overwrite, have to modify like https://github.com/Barry-Xu-2018/rcl/commit/871601ebb24f5ad4a310251b57e310d9412840ee